src/main/java/org/dita/dost/reader/DitaValReader.java

```diff
@@ -193,7 +193,8 @@ public void readProp(final Element elem)  {
                 action = readFlag(elem);
                 break;
             default:
-                throw new IllegalArgumentException("Invalid action: " + attAction);
+                logger.error(MessageUtils.getMessage("DOTJ077F", attAction).toString());
```

Comment:
Instead of logging an error and then throwing an unchecked exception, this should just throw the exception with the message.

Reply:
My initial thought was that if we had it as a message, we could push some additional info into the message guide along with it - for example, "The valid values are x / y / z, this message may also indicate a typo or extra space in the value" - but I don't think we'd be able to provide that info if it's just part of the exception? Well - we could, but I know there's generally an effort not to leave troubleshooting / diagnostic information to the user guide.

Reply:
The error messages should be as terse as possible and all the educational info should go into the docs.

Reply:
If you want to add debug info, then stick that debug into into a debug log message, but the `DOTJ077F` error message should be thrown in the `IllegalArgumentException` because that's what will be in the stack trace and the final message.

Reply:
I misunderstood the suggestion (I think I meant something different by debug information) -- fixing now as suggested and will upload shortly.